@here/harp-mapview/lib/MapViewAtmosphere.ts

```diff
@@ -0,0 +1,481 @@
+        if (this.m_groundMaterial !== undefined) {
+            const groundMat = this.m_groundMaterial as GroundAtmosphereMaterial;
+            groundMat.setDynamicLighting(dynamicLight);
+        }
+        if (this.m_skyMaterial !== undefined) {
+            const skyMat = this.m_skyMaterial as SkyAtmosphereMaterial;
+            skyMat.setDynamicLighting(dynamicLight);
+        }
+    }
+
+    /**
+     * Disposes allocated resources.
+     */
+    dispose() {
+        // Unlink from scene and mapview anchors
+        if (this.enabled) {
+            this.enabled = false;
+        }
+
+        this.m_skyMaterial?.dispose();
+        this.m_groundMaterial?.dispose();
+
+        this.m_skyGeometry?.dispose();
+        this.m_groundGeometry?.dispose();
+
+        // After disposal we may no longer enable effect.
+        this.m_skyGeometry = undefined;
+        this.m_groundGeometry = undefined;
+
+        this.m_skyMaterial = undefined;
+        this.m_groundMaterial = undefined;
+
+        this.m_skyMesh = undefined;
+        this.m_groundMesh = undefined;
+    }
+
+    /**
+     * Sets the atmosphere depending on the [[Theme]] instance provided.
+     *
+     * This function is called when a theme is loaded. Atmosphere is added only if the theme
+     * contains a atmosphere definition with a:
+     * - `color` property, used to set the atmosphere color.
+     *
+     * @param theme A [[Theme]] instance.
+     */
+    reset(theme: Theme) {
+        //this.m_cachedTheme = theme;
+    }
+
+    /**
+     * Updates the atmosphere at runtime, depending on the camera and projection settings.
+     *
+     * @param camera An instance of a `THREE.Camera`.
+     * @param projection An instance of Projection currently in use.
+     */
+    update(camera: THREE.PerspectiveCamera | THREE.OrthographicCamera, projection: Projection) {
+        assert(
+            projection.type === this.m_mapView.projection.type,
+            "Projection type changed, please re-create MapViewAtmosphere"
+        );
+    }
+
+    private get disposed() {
+        return this.m_skyMesh === undefined && this.m_groundMesh === undefined;
+    }
+
+    /**
+     * Handles atmosphere addition.
+     */
+    private addToScene(scene: THREE.Scene) {
+        assert(!MapViewAtmosphere.isPresent(scene), "Atmosphere already added");
+        if (this.m_skyMesh !== undefined) {
+            scene.add(this.m_skyMesh);
+        }
+        if (this.m_groundMesh !== undefined) {
+            scene.add(this.m_groundMesh);
+        }
+    }
+
+    /**
+     * Handles atmosphere removal.
+     */
+    private removeFromScene(scene: THREE.Scene) {
+        if (MapViewAtmosphere.isPresent(scene)) {
+            if (this.m_skyMesh !== undefined) {
+                scene.remove(this.m_skyMesh);
+            }
+            if (this.m_groundMesh !== undefined) {
+                scene.remove(this.m_groundMesh);
+            }
+        }
+    }
+
+    private addToMapView(mapView: MapView) {
```

Comment:
I'm confused, do we use map anchors or do we add the meshes to the scene?

Reply:
Fixed now.